candle-metal-kernels/src/ternary.metal

```diff
@@ -34,9 +34,18 @@ METAL_FUNC void where_cond(
     if (i >= numel){
        return;
     }
+
+    bool strides_match = true;
```

Comment:
I'm not sure that would actually be correct. What you want to check is that the tensors are in C layout because that's what's expected from the output tensors at the moment. If you look at the cuda kernels, that's what we would test for e.g. [here](https://github.com/huggingface/candle/blob/143c481c20abc3420e848eab075d1547a96cc447/candle-kernels/src/ternary.cu#L18).